src/components/AppBreadcrumbs/AppBreadcrumbs.tsx

```diff
@@ -0,0 +1,73 @@
+import React from 'react';
+import { withStyles } from '@material-ui/core/styles';
+import { Breadcrumbs, Link } from '@material-ui/core';
+import { Link as RouterLink } from 'react-router-dom';
+
+function capitalize(str: string): string {
+  return str.charAt(0).toUpperCase() + str.slice(1);
+}
+
+const styles = (theme: any) => ({
+  terminalBreadcrumb: {
+    color: theme.palette.primary.dark,
+    cursor: 'default',
+    '&:hover': {
+      textDecoration: 'none',
+    },
+  },
+});
+
+export type Context = { type: string; id: string; name: string };
+export type Breadcrumb = { text: string; to: string };
+
+type AppBreadcrumbsProps = {
+  childBreadcrumbs: Array<Breadcrumb>;
+  context: Context;
+};
+
+const BreadcrumbLink = withStyles(styles)((props: any) => {
+  const { classes, ...other } = props;
+  if (props.disabled) {
+    return (
+      <Link
+        component={RouterLink}
+        color="primary"
+        className={classes.terminalBreadcrumb}
+        onClick={(e) => e.preventDefault()}
+        {...other}
+      />
+    );
+  }
+  return <Link component={RouterLink} color="primary" {...other} />;
+});
```

Comment:
I don't know that it makes sense to pull this into a separate constant within the same file. If it's not shared with anything, I think it should live within the AppBreadcrumbs component. If it needs to be a separate, reusable component, then it should live in its own file.

Reply:
Here's where my understanding of scoping in Javascript reaches its limits. My assumption is that, if the function is in a file and not `export`ed, then its scope is confined to the file. For me, it makes sense for this component to be a separate function in the same file for organization and readability reasons.

Reply:
I'm okay with keeping it as it, but a few thoughts:
- My understanding of React conventions is what I stated before: This - which is definitely more a small component than just a function - should either live in its own file or within the default component of the page. I think there is some merit to sticking to conventions, but they're not always the best
- I do not find this approach more readable - the additional decoration of the function withStyles plus having to track a second list of props... which has the type of any! (the horror!) makes this seem more complex than it needs to be.

Reply:
Good points. Changed so that `BreadcrumbLink` is in its own file!